### PR TITLE
Use duration input for timeout in wait_for_trigger action

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Use the duration input for the timeout field of the wait for trigger action (currently, it is a text field).

CleanShot 2022-08-20 at 21 31 13

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts [hunk 1]
```diff
@@ -52,6 +55,17 @@ export class HaWaitForTriggerAction
...
+      return;
+    }
+    fireEvent(this, "value-changed", {
+      value: { ...this.action, timeout: value },
+    });
+  }
+
   private _continueChanged(ev) {
     fireEvent(this, "value-changed", {
       value: { ...this.action, continue_on_timeout: ev.target.checked },
```
Comment (steverep) [inline src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts:71]: You can use `CustomEveent<DetailType>` to better type this.

Comment (frenck) [reply src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts:71]: That would be changing existing code. As this code isn't changed or touched in this PR.

File: src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts [hunk 2]
```diff
@@ -52,6 +55,17 @@ export class HaWaitForTriggerAction
     `;
   }
 
+  private _timeoutChanged(ev: CustomEvent): void {
```
Comment (steverep) [inline src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts:58]: That comment was supposed to be for this line on typing `ev`.  Commenting on lines in the mobile app is horrible with VoiceOver.

The diff is showing the whole method was added.  Is that not right?

**Review by steverep — DISMISSED** [review]

File: src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts (hunk 2 above)
Comment (frenck) [reply src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts]: > Commenting on lines in the mobile app is horrible with VoiceOver.

Uh comments on larger diff sometimes look horrible from the UI as well (it looked like the comment was on the part below).

Anyways, yeah, understood and agree 👍 

I've pushed a change to make the typing more specific.

**Review by steverep — APPROVED** [review]

**Pull request merged**